app/src/main/java/com/lagradost/cloudstream3/metaproviders/TraktProvider.kt

```diff
@@ -0,0 +1,585 @@
+                this.rating = mediaDetails.rating?.times(1000)?.roundToInt()
+                this.tags = mediaDetails.genres
+                this.duration = mediaDetails.runtime
+                addTrailer(mediaDetails.trailer)
+                this.recommendations = relatedMedia
+                this.actors = actors
+                //comingSoon
+                //posterHeaders
+                this.backgroundPosterUrl = backDropUrl
+                this.contentRating = mediaDetails.certification
+            }
+        }
+    }
+    private suspend fun getApi(url: String) : String {
+        return app.get(
+            url = url,
+            headers = mapOf(
+                "Content-Type" to "application/json",
+                "trakt-api-version" to "2",
+                "trakt-api-key" to traktClientId,
+            )
+        ).toString()
+    }
+
+    private fun getPosterUrl(images: Images, random: Boolean) : String? {
+
+        val imagePath = when {
+            random -> {
+                images.posters?.filter { it.iso6391 == "en" }?.randomOrNull() ?: images.posters?.randomOrNull()
+            }
+            else -> {
+                images.posters?.firstOrNull { it.iso6391 == "en" } ?: images.posters?.firstOrNull()
+            }
+        }
+            ?: return null
+        return "https://image.tmdb.org/t/p/w500${imagePath.filePath}"
+
+    }
+
+    private fun getOrigPosterUrl(images: Images, random: Boolean) : String? {
+
+        val imagePath = when {
+            random -> {
+                images.posters?.filter { it.iso6391 == null }?.randomOrNull() ?: images.posters?.randomOrNull()
+            }
+            else -> {
+                images.posters?.firstOrNull { it.iso6391 == null } ?: images.posters?.firstOrNull()
+            }
+        }
+            ?: return null
+        return "https://image.tmdb.org/t/p/original${imagePath.filePath}"
+    }
+
+    private fun getBackdropUrl(images: Images) : String? {
+        val imagePath = images.backdrops?.filter { it.iso6391 == null }?.randomOrNull() ?: images.backdrops?.randomOrNull()
+        ?: return null
+        return "https://image.tmdb.org/t/p/original${imagePath.filePath}"
+    }
+
+    private fun getImageUrl(path: String?) : String? {
+        if (path == null) return null
+        return "https://image.tmdb.org/t/p/w500${path}"
+    }
+
+    private suspend fun getProfileImages(castTmdbId: String, movie: Boolean) : Images {
+        val imageApiUrl = if (movie) {
+            "$tmdbApiUrl/movie/$castTmdbId/credits?api_key=$tmdbApiKey"
+        } else {
+            "$tmdbApiUrl/tv/$castTmdbId/credits?api_key=$tmdbApiKey"
+        }
+
+        val req = app.get(
+            url = imageApiUrl,
+            headers = mapOf(
+                "accept" to "application/json",
+            )
+        ).toString()
+        return parseJson<Images>(req)
+    }
+
+    private suspend fun getImages(tmdbId: String, movie: Boolean = true) : Images {
```

Comment:
👎 This is really really bad, we never want to make one get request for every search response as that more or less creates 100 requests for 1 search instead of 1. Is there any way to get the image of a searchresponse without making an additional request?